vulkan_generator/generator.py

```diff
@@ -57,8 +61,32 @@ def print_vulkan_metaadata(vulkan_metadata: types.AllVulkanTypes) -> None:
     pretty_printer.pprint(vulkan_metadata.funcpointers)
 
 
-def generate(vulkan_xml_path: Path) -> bool:
+def basic_generate(target: str,
+                   output_dir: Path,
+                   all_vulkan_types: types.AllVulkanTypes,
+                   generate_header,
+                   generate_cpp,
+                   generate_test):
+
+    generate_header(os.path.join(output_dir, target + ".h"), all_vulkan_types)
+    generate_cpp(os.path.join(output_dir, target + ".cpp"), all_vulkan_types)
+    generate_test(os.path.join(output_dir, target + "_tests.cpp"), all_vulkan_types)
```

Comment:
should be `.cc`, names need to be all lowercase and with underscore, and tests should be in `_test.cc` according to the style guide (e.g. `handle_remapper.h`, `handle_remapper.cc`, and `handle_remapper_test.cc`).

Reply:
https://google.github.io/styleguide/cppguide.html#File_Names